### PR TITLE
XDR for upgradeable config entries

### PR DESCRIPTION
**Pull request by sisuresh**
This contains changes from https://github.com/stellar/stellar-xdr/pull/27 and a change for upgrading config entries for https://github.com/stellar/stellar-core/pull/3674.

### CODE REVIEW

**Review by dmkozh — APPROVED** [review]

File: Stellar-ledger-entries.x [hunk 1]
```diff
@@ -518,20 +518,13 @@ case CONFIG_SETTING_TYPE_UINT32:
```
Comment (jayz22) [inline Stellar-ledger-entries.x]: Is the `ConfigSetting` still needed? I see the `ConfigSettingEntry` below is using `uint32` directly so maybe this is obsolete?

Comment (dmkozh) [reply Stellar-ledger-entries.x]: Good catch, we indeed don't use this anymore.

Comment (sisuresh) [reply Stellar-ledger-entries.x]: Yeah good catch. Removed.

File: Stellar-ledger.x [hunk 2]
```diff
@@ -130,6 +130,11 @@ enum LedgerUpgradeType
     LEDGER_UPGRADE_CONFIG = 6
 };
 
+struct ConfigUpgradeSetKey {
+    Hash contractID;
+    Hash contentHash;
+};
+
```
Comment (jayz22) [inline Stellar-ledger.x:137]: If I'm understanding correctly, the `ConfigUpgradeSetKey` here must be a contract data belong to `ContractID`? Does that mean only a ContractData entry can be upgraded? 
I'm thinking in the context of network-wide setting such as metering cost params that wouldn't belong to a contract, how will it work?

Comment (sisuresh) [reply Stellar-ledger.x:137]: The actual `ConfigSettingEntry` is not associated with a contract and is not upgraded in Soroban. We are just adding a mechanism where you can look up the xdr for a new `ConfigSettingEntry` in `ContractData`, and update the existing `ConfigSettingEntry` to use that.

**Pull request merged**